plugin-server/src/worker/ingestion/event-pipeline/runner.ts

```diff
@@ -193,6 +193,23 @@ export class EventPipelineRunner {
             return this.registerLastStep('clientIngestionWarning', [event], kafkaAcks)
         }
 
+        // Track $set usage in events that aren't known to use it, before plugins which add various $set props.
+        const person_events = ['$set', '$identify', '$create_alias', '$merge_dangerously', '$groupidentify']
+        const known_set_events = [
+            '$feature_interaction',
+            '$feature_enrollment_update',
+            'survey dismissed',
+            'survey sent',
+        ]
```

Comment:
Let's make these [Sets](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) for faster lookup, and set them as global constant at the top of the file to avoid re-creating them everytime.

Reply:
will do, I didn't realize something like that makes a difference